addon/compress.h

```diff
@@ -0,0 +1,21 @@
+#include <vector>
+
+#include "compression_worker.h"  // CompressionResult
+#include "zstd.h"
+
+CompressionResult compress(const std::vector<uint8_t> data, size_t compression_level) {
+    size_t output_buffer_size = ZSTD_compressBound(data.size());
+    std::vector<uint8_t> output(output_buffer_size);
+
+    size_t result_code =
+        ZSTD_compress(output.data(), output.size(), data.data(), data.size(), compression_level);
```

Comment:
Why did we decide not to use the streaming compression API here, but used the streaming decompression API?